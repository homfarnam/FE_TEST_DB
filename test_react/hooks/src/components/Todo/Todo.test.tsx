import { render, screen } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from 'react-query'
import { Todo } from '.'
import * as hooks from '../../hooks/json.hooks'

const client = new QueryClient({
  defaultOptions: { queries: { retry: false } },
})
const Wrapper: React.FC = ({ children }) => {
  return <QueryClientProvider client={client}>{children}</QueryClientProvider>
}

jest.mock('../../hooks/json.hooks', () => ({
  useGetTodo: jest.fn(),
}))

const getTodo = hooks.useGetTodo as jest.Mocked<any>

beforeEach(() => {
  jest.clearAllMocks()
  client.clear()
})

test('render <Todo /> when loading data', async () => {
  getTodo.mockReturnValue({
    isLoading: true,
  })
  const { container } = render(<Todo id={1} />, { wrapper: Wrapper })

  expect(screen.getByText(/loading/i)).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})

test('render <Todo /> after getting data successfully', () => {
  const todo = {
    userId: 1,
    id: 1,
    title: 'delectus aut autem',
    completed: false,
  }
  getTodo.mockReturnValue({ isSuccess: true, data: todo })

  const { container } = render(<Todo id={1} />, { wrapper: Wrapper })

  expect(container).toMatchSnapshot()
})

test('render <Todo /> after getting data failed', async () => {
  getTodo.mockReturnValue({ isError: true })

  const { container } = render(<Todo id={1} />, { wrapper: Wrapper })

  expect(container).toMatchSnapshot()
})
