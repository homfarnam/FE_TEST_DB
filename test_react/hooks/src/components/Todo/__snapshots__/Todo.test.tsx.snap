// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Todo /> after data has been fetched 1`] = `
<div>
  <div>
    <h4>
      delectus aut autem
    </h4>
    <p>
      ID: $
      1
    </p>
    <p>
      Completed: 
      no
    </p>
    <p>
      user: 
      1
    </p>
  </div>
</div>
`;

exports[`render <Todo /> after getting data failed 1`] = `
<div>
  <p
    role="alert"
  >
    failed to load todo
  </p>
</div>
`;

exports[`render <Todo /> after getting data successfully 1`] = `
<div>
  <div>
    <h4>
      delectus aut autem
    </h4>
    <p>
      ID: $
      1
    </p>
    <p>
      Completed: 
      no
    </p>
    <p>
      user: 
      1
    </p>
  </div>
</div>
`;

exports[`render <Todo /> when loading data 1`] = `
<div>
  <p
    aria-label="loading"
  >
    loading...
  </p>
</div>
`;
